### ISSUE TITLE
non-mixer-xt hnags on Fedora 40

### ISSUE THREAD

**Issue by ycollet** [Stazed/nonlib-xt#3]
I am using the lastest release of non-mixer-xt on fedora 40.
I start non-mixer-xt manually and try to add 3 strips via the "add N strips" menu.
When I click on OK; non-mixer-xt hangs.
Here is the stack trace:
```
#0  __pthread_kill_implementation (threadid=<optimized out>, signo=signo@entry=6, no_tid=no_tid@entry=0) at pthread_kill.c:44
#1  0x00007ffff76ae8a3 in __pthread_kill_internal (signo=6, threadid=<optimized out>) at pthread_kill.c:78
#2  0x00007ffff765c8ee in __GI_raise (sig=sig@entry=6) at ../sysdeps/posix/raise.c:26
#3  0x00007ffff76448ff in __GI_abort () at abort.c:79
#4  0x00007ffff78d95d0 in std::__glibcxx_assert_fail (file=file@entry=0x5555556a29b0 "/usr/include/c++/13/bits/stl_vector.h", line=line@entry=1128, 
    function=function@entry=0x5555556a28e0 "std::vector<_Tp, _Alloc>::reference std::vector<_Tp, _Alloc>::operator[](size_type) [with _Tp = Module::Port; _Alloc = std::allocator<Module::Port>; reference = Module::Port&; size_type = long unsigne"..., condition=condition@entry=0x5555556a22ca "__n < this->size()") at ../../../../../libstdc++-v3/src/c++11/assert_fail.cc:41
#5  0x0000555555605223 in std::vector<Module::Port, std::allocator<Module::Port> >::operator[] (__n=<optimized out>, this=<optimized out>) at /usr/include/c++/13/bits/stl_vector.h:992
#6  Module::control_input_port_index (p=<optimized out>, this=<optimized out>) at /usr/src/debug/non-mixer-xt-2.0.1-1.fc39.x86_64/mixer/src/Module.H:637
#7  Module_Parameter_Editor::handle_control_changed (p=<optimized out>, this=<optimized out>) at /usr/src/debug/non-mixer-xt-2.0.1-1.fc39.x86_64/mixer/src/Module_Parameter_Editor.C:1061
#8  Module::handle_control_changed (this=<optimized out>, p=0x7fffffffc880) at /usr/src/debug/non-mixer-xt-2.0.1-1.fc39.x86_64/mixer/src/Module.C:496
#9  0x000055555559b275 in Module::Port::control_value (this=this@entry=0x7fffffffc880, f=<optimized out>) at /usr/src/debug/non-mixer-xt-2.0.1-1.fc39.x86_64/mixer/src/Module.H:368
#10 0x000055555559dba8 in Gain_Module::Gain_Module (this=<optimized out>, this=<optimized out>) at /usr/src/debug/non-mixer-xt-2.0.1-1.fc39.x86_64/mixer/src/Gain_Module.C:40
#11 0x00005555555fb5c5 in Chain::initialize_with_default (this=0x55555573ff70) at /usr/src/debug/non-mixer-xt-2.0.1-1.fc39.x86_64/mixer/src/Chain.C:287
#12 Mixer_Strip::Mixer_Strip (strip_name=<optimized out>, this=0x5555557f4290) at /usr/src/debug/non-mixer-xt-2.0.1-1.fc39.x86_64/mixer/src/Mixer_Strip.C:92
#13 Mixer::new_strip (this=this@entry=0x55555574cf30) at /usr/src/debug/non-mixer-xt-2.0.1-1.fc39.x86_64/mixer/src/Mixer.C:1138
#14 0x00005555555fc088 in Mixer::command_add_strip (this=0x55555574cf30) at /usr/src/debug/non-mixer-xt-2.0.1-1.fc39.x86_64/mixer/src/Mixer.C:1522
#15 Mixer::cb_menu (this=0x55555574cf30, o=<optimized out>) at /usr/src/debug/non-mixer-xt-2.0.1-1.fc39.x86_64/mixer/src/Mixer.C:333
#16 0x000055555566b46a in Fl_Widget::do_callback(Fl_Widget*, void*) ()
#17 0x000055555564dd78 in Fl_Menu_::picked(Fl_Menu_Item const*) ()
#18 0x000055555564e21d in Fl_Menu_Bar::handle(int) ()
#19 0x0000555555641d5b in Fl_Group::handle(int) ()
#20 0x0000555555641d5b in Fl_Group::handle(int) ()
#21 0x00005555555f773b in Mixer::handle (this=0x55555574cf30, m=1) at /usr/src/debug/non-mixer-xt-2.0.1-1.fc39.x86_64/mixer/src/Mixer.C:1240
#22 0x0000555555641d5b in Fl_Group::handle(int) ()
#23 0x0000555555632d15 in send_event(int, Fl_Widget*, Fl_Window*) ()
#24 0x0000555555634fbb in Fl::handle_(int, Fl_Window*) [clone .part.0] ()
#25 0x0000555555671a7b in fl_handle(_XEvent const&) ()
#26 0x00005555556730a8 in do_queued_events() ()
#27 0x00005555556733f9 in fl_wait(double) ()
#28 0x0000555555633a39 in Fl::wait(double) ()
#29 0x0000555555633af9 in Fl::run() ()
#30 0x0000555555587141 in main (argc=<optimized out>, argv=<optimized out>) at /usr/src/debug/non-mixer-xt-2.0.1-1.fc39.x86_64/mixer/src/main.C:378
```

**Comment by Stazed** [comment]
I don't see anything from the trace that would shed light on why this is happening. It looks like a memory allocation problem. 

I assume you are using pipewire with Fedora 40. Is there any chance you can try with jack2? 

My personal experience with pipewire jack is that it is completely unusable for me, and getting worse...

**Comment by ycollet** [comment]
non-mixer-xt was built using the following flags:
```
-O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64 -march=x86-64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer 
```

Add I use non-mixer-xt with pipewire-jack

**Comment by ycollet** [comment]
One of this flag activate some exceptions in standard libs.
The main message I've got:
```
/usr/include/c++/14/bits/stl_vector.h:1130: std::vector<_Tp, _Alloc>::reference std::vector<_Tp, _Alloc>::operator[](size_type) [with _Tp = Module::Port; _Alloc = std::allocator<Module::Port>; reference = Module::Port&; size_type = long unsigned int]: Assertion '__n < this->size()' failed.
Abandon (core dumped)
```
Probably a problem related to accessing an empty stl vector.

**Comment by ycollet** [comment]
```
#6  0x00005555555f0050 in Module::handle_control_changed (this=0x55555572ca50, p=0x7fffffffc2f0)
    at /usr/src/debug/non-mixer-xt-2.0.1-2.fc40.x86_64/mixer/src/Module.C:499
499         if (bypassable() && p == &control_input[control_input.size() - 1])
(gdb) print control_input.size()
$1 = 0
```

**Comment by Stazed** [comment]
O.K. got it!!!
Will have a fix out shortly. Just need to determine the best way to handle it without side effects.
Thanks for narrowing it down!

**Comment by ycollet** [comment]
Thanks a lot for your work !

**Comment by Stazed** [comment]
Just pushed a fix to non-mixer-xt wip branch. Seems to be working with your build options now and I don't think it will break anything else. Let me know if it does not work for you.

**Comment by ycollet** [comment]
Tested and it worked.
Maybe this will fix #21 ?
